### PR TITLE
Cluster autoscaler test

### PR DESCRIPTION
**Pull request by whites11**
This PR adds a new e2e test for the cluster autoscaling feture.

4 tests are failing because I didn't run them.

### CODE REVIEW

File: e2e/test/clusterautoscaler/.env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+TEST_DIR="e2e/test/clusterautoscaler"
+AZURE_AZS=1
+KEEP_RESOURCES=true
```
Comment (fiunchinho) [inline e2e/test/clusterautoscaler/.env:3]: I guess you don't need to keep the resources after you made sure the code it's fine.

Comment (whites11) [reply e2e/test/clusterautoscaler/.env]: Good catch, thanks

File: e2e/test/clusterautoscaler/provider.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	}
+	if config.ClusterID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterID must not be empty", config)
+	}
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+
+	p := &Provider{
+		azureClient: config.AzureClient,
```
Comment (fiunchinho) [inline e2e/test/clusterautoscaler/provider.go:49]: I believe you don't need `azureClient`, `clusterID` or `ctrlClient`.

File: e2e/test/clusterautoscaler/main_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			panic(microerror.JSON(err))
+		}
+	}
+
+	{
+		c := Config{
+			Logger:          config.Logger,
+			Guest:           config.Guest,
+			Provider:        p,
+			ClusterID:       env.ClusterID(),
```
Comment (fiunchinho) [inline e2e/test/clusterautoscaler/main_test.go:49]: I believe you don't need `ClusterID` or `TargetNamespace`

**Review by fiunchinho — APPROVED** [review]
This is G-R-E-A-T. Really good job, man. Btw, not sure why pkged.go got updated :joy:

**Comment by whites11** [comment]
> This is G-R-E-A-T. Really good job, man. Btw, not sure why pkged.go got updated

Nobody knows :D

File: e2e/test/clusterautoscaler/provider.go (hunk 2 above)
Comment (whites11) [reply e2e/test/clusterautoscaler/provider.go]: removed

File: e2e/test/clusterautoscaler/main_test.go (hunk 3 above)
Comment (whites11) [reply e2e/test/clusterautoscaler/main_test.go:46]: removed

**Pull request merged**